examples/security-token/assets/controller.py

```diff
@@ -0,0 +1,222 @@
+    assetManager = AssetParam.manager(Int(0))
+
+    on_deployment = Seq([
+        assetManager, # load asset manager from store
+		Assert(And(
+			Txn.application_args.length() == Int(1),
+			# Txn.assets.length() == Int(1), [TEALv3]
+			no_rekey_addr,
+
+            # Controller should be deployed by ASA.manager
+            assetManager.hasValue(),
+			Txn.sender() == assetManager.value()
+        )),
+
+        # total permission contracts during deploy is set to 0.
+        App.globalPut(var_total, Int(0)),
+
+        # set kill_status to false(0) during deploy
+        App.globalPut(var_is_killed, Int(0)),
+
+        # Save token index in controller's global state
+        App.globalPut(token_id, Btoi(Txn.application_args[0])), # or maybe use App.globalPut(token_id, Btoi(Txn.assets[0])) with TEALv3
+        Return(Int(1))
+    ])
+
+    # retreive asset reserve from Txn.ForeignAssets[0]
+    assetReserve = AssetParam.reserve(Int(0))
+    issue_token = Seq([
+        assetReserve,
+        Assert(And(
+			Txn.application_args.length() == Int(1),
+			# Txn.assets.length() == Int(1), [TEALv3]
+            Gtxn[1].type_enum() == TxnType.AssetTransfer,
+            Gtxn[1].xfer_asset() == App.globalGet(token_id), # verify if token index is correct
+
+            # Issue should only be done by token reserve
+            assetReserve.hasValue(),
+			Gtxn[0].sender() == assetReserve.value(),
+            Gtxn[1].asset_sender() == assetReserve.value(),
+
+            # only allow issue if token is not killed
+            App.globalGet(var_is_killed) == Int(0)
+        )),
+        Return(Int(1))
+    ])
+
+    # kill a token, after verifying transaction set is_killed to true(Int(1))
+    assetManager = AssetParam.manager(Int(0))
+    kill_token = Seq([
+        assetManager, # load asset_manager of Txn.ForeignAssets[0]
+        Assert(And(
+			Txn.application_args.length() == Int(1),
+			# Txn.assets.length() == Int(1), [TEALv3],
+            # Txn.assets[0] == App.globalGet(token_id), [TEALv3] # verify if token index is correct
+            Txn.type_enum() == TxnType.ApplicationCall,
+
+            # Only asset manager can kill the token
+            assetManager.hasValue(),
+			Txn.sender() == assetManager.value(),
+        )),
+
+        # finally set is_killed to true
+        App.globalPut(var_is_killed, Int(1)),
+        Return(Int(1))
+    ])
+
+    add_new_permission = Seq([
+        assetManager,
+        Assert(And(
+            Txn.application_args.length() == Int(3),
+            # Txn.assets[0] == App.globalGet(token_id), [TEALv3]
+
+            assetManager.hasValue(),
+			Txn.sender() == assetManager.value(),
+        )),
+
+        # Update total rules counter (= 1 in this case) as we only have a single
+        # permissions contract
+        App.globalPut(var_total, App.globalGet(var_total) + Int(1)),
+
+        # Add permissions(rules) smart contract config in global state (id, manager)
```

Comment:
please add what is `id` here ? is it smart contract id ?

Reply:
`permissions_id`, `permissions_manager`. updated.